core/iwasm/common/wasm_shared_memory.c

```diff
@@ -87,6 +92,53 @@ search_module(WASMModuleCommon *module)
     return NULL;
 }
 
+static void
+create_list_of_waiter_addresses(void *key, void *value, void *user_data)
+{
+    AtomicWaitAddressArgs *data = (AtomicWaitAddressArgs *)user_data;
+    if (data->len > 0) {
+        if (!(data->addr = wasm_runtime_realloc(data->addr,
```

Comment:
When to free the memory? And why do nothing except LOG_ERROR when malloc/realloc memory failed?